numpy/core/tests/test_multiarray.py

```diff
@@ -5047,6 +5047,19 @@ def test_fromfile_offset(self):
                     np.fromfile, self.filename, dtype=self.dtype,
                     sep=",", offset=1)
 
+    def test_fromfile_bad_dup(self):
+        def dup_str(fd):
+            return 'abc'
+
+        def dup_bigint(fd):
+            return 2*68
+        
+        old_dup = os.dup
+        with open(self.filename, 'wb') as f:
+            self.x.tofile(f)
+            for dup in (dup_str, dup_bigint):
+                assert_raises(OSError, np.fromfile, f)
```

Comment:
```suggestion
        old_dup = os.dup
        try:
            with open(self.filename, 'wb') as f:
                self.x.tofile(f)
                for dup in (dup_str, dup_bigint):
                    os.dup = dup
                    assert_raises(OSError, np.fromfile, f)
        finally:
            os.dup = old_dup
```
The monkeypatching part of the test seems to be missing? (I am a bit surprised not more tests are failing though).

Reply:
thanks